hs_core/hydroshare/utils.py

```diff
@@ -679,19 +688,36 @@ def resource_pre_create_actions(resource_type, resource_title, page_redirect_url
     from.resource import check_resource_type
     from hs_core.views.utils import validate_metadata
 
+    resource_cls = check_resource_type(resource_type)
+    size = 0
+    if len(files) > 0:
+        size = validate_resource_file_size(files)
+        validate_resource_file_count(resource_cls, files)
+        validate_resource_file_type(resource_cls, files)
+
+    if requesting_user:
+        # validate it is within quota hard limit
```

Comment:
Would be good to put this validation code as a separate function as you have used the same code multiple places.